### PR TITLE
Photon.js: add unique constraints

### PR DESCRIPTION
**Pull request by matthewmueller**
Add unique constraints spec

By the way, this was done sorta quickly and the current code blocks make it hard to add this information in a sane way. Ideally we can turn this into a real spec.

### CODE REVIEW

File: photonjs/README.md [hunk 1]
```diff
@@ -208,6 +214,119 @@ const deletedCount: number = await photon.user.delete().count()
...
+
+### Naming the composite constraint
+
+```groovy
+model User {
+    id String @id
+    firstName String
+    lastName String
+    email String @unique
+    @@unique([ firstName, lastName ], name: "fullName")
```
Comment (mavilein) [inline photonjs/README.md]: We have already implemented the `name` argument but with different semantics. Currently the `name` argument refers to the name of the underlying unique index in the database. In your example it insteads influences the name of the combined criterion in the Photon API. I think we should distinguish those two names as they are different concepts, e.g. `dbName` vs `photonName` (not good i know. just to illustrate my point)
The reason we need different names is that the name of the underlying index must be unique in some databases e.g. Postgres.  And i think we do want to support configuring the index name.

@matthewmueller : I just discovered this by interpreting your desired semantics. We should clearly document what the arguments for each attribute mean like in the datamodel v1.1 [spec](https://github.com/prisma/prisma/issues/3405).

Comment (matthewmueller) [reply photonjs/README.md]: Ah okay that makes sense! I just realized it's also in here: https://github.com/prisma/specs/tree/master/schema#core-block-attributes but that could use some clarification.

Comment (matthewmueller) [reply photonjs/README.md]: I agree that we should distinguish, I'll think about some options there 👍

Comment (mavilein) [reply photonjs/README.md]: Just discovered that this is the photon spec here. So my last sentence does not make sense here as it applies to the PSL spec.

File: photonjs/README.md [hunk 2]
```diff
@@ -208,6 +214,119 @@ const deletedCount: number = await photon.user.delete().count()
 uts
 ```
 
+## Field-level Primary Key constraint
+
+Different unique constraints will change Photon's `where` blocks:
+
+````groovy
```
Comment (janpio) [inline photonjs/README.md]:
```suggestion
```groovy
```

File: photonjs/README.md (hunk 1 above)
Comment (matthewmueller) [reply photonjs/README.md]: I've renamed this to `alias`.

File: photonjs/README.md [hunk 3]
```diff
@@ -192,20 +198,129 @@ await photon.user.findMany({
...
 
 // Simple aggregation short
 // TODO more examples
 const deletedCount: number = await photon.user.delete().count()
-uts
+```
+
+## Field-level Primary Key constraint
+
+Different unique constraints will change Photon's `where` blocks:
```
Comment (janpio) [inline photonjs/README.md]: `where` block is the thing inside `find(...)`?

**Comment by janpio** [comment]
I read through the diff:
- It changes `findUnique` to `find` - is this related to this "unique constraints"?
- I assume the other formatting changes (multi line to single line per command) are unrelated?
- https://github.com/prisma/specs/pull/179/files#diff-e74be82aecbeb981e879b47b9e886b95R212-R323 and https://github.com/prisma/specs/pull/179/files#diff-e74be82aecbeb981e879b47b9e886b95R466-R477 are the actual unique contraint changes, right?

File: photonjs/README.md (hunk 3 above)
Comment (matthewmueller) [reply photonjs/README.md]: yep

**Pull request merged**